In [1]:
# OPEN,HIGH,LOW,CLOSE,TOTTRDQTY,TOTTRDVAL,TIMESTAMP
import scipy
import numpy as np
import pandas as pd

In [2]:
# open SOUTHBANK.csv
df = pd.read_csv('SOUTHBANK.csv')

In [3]:
print(df.head())

     OPEN    HIGH     LOW   CLOSE  TOTTRDQTY     TOTTRDVAL    TIMESTAMP
0  207.50  213.30  205.40  212.10     701915  1.473843e+08  19-AUG-2010
1  212.00  212.00  208.00  208.45     229254  4.798819e+07  20-AUG-2010
2  209.90  210.00  204.00  204.80     269180  5.530275e+07  23-AUG-2010
3  206.30  206.80  202.05  205.05     240741  4.936476e+07  24-AUG-2010
4  205.95  205.95  199.00  201.10     242027  4.860458e+07  25-AUG-2010


In [4]:
import pandas as pd
import numpy as np
from scipy.stats import norm

# Load the dataset
file_path = 'SOUTHBANK.csv'  # Update the path if necessary
southbank_data = pd.read_csv(file_path)

# Step 1: Extract the CLOSE price column and sort it
close_prices = southbank_data['CLOSE'].sort_values()

close_prices.head(40)

121    18.30
120    18.35
757    19.25
756    19.30
119    19.35
752    19.45
754    19.55
117    19.60
122    19.60
753    19.60
781    19.65
751    19.70
780    19.70
828    19.70
782    19.70
118    19.75
779    19.75
755    19.75
774    19.80
775    19.80
778    19.80
864    19.80
826    19.85
776    19.85
869    19.85
114    19.85
115    19.90
859    19.90
867    19.90
825    19.90
116    19.95
865    19.95
777    19.95
257    19.95
868    20.00
861    20.00
856    20.00
870    20.00
855    20.00
860    20.00
Name: CLOSE, dtype: float64

In [5]:
# Step 2: Calculate the median of the CLOSE prices
median_close = close_prices.median()

# Step 3: Identify the 25% observations on either side of the median
lower_bound = np.percentile(close_prices, 25)  # 25th percentile
upper_bound = np.percentile(close_prices, 75)  # 75th percentile

# Step 4: Filter the CLOSE prices within this range
sample_close_prices = close_prices[(close_prices >= lower_bound) & (close_prices <= upper_bound)]

lower_bound, upper_bound

(np.float64(22.2), np.float64(25.95))

In [6]:
# Step 5: Calculate the sample mean
sample_mean = sample_close_prices.mean()

# Step 6: Calculate the standard error of the mean (SEM)
sample_std = sample_close_prices.std()
sample_size = len(sample_close_prices)
sem = sample_std / np.sqrt(sample_size)

# Step 7: Calculate the 90% Confidence Interval
confidence_level = 0.90
z_score = norm.ppf((1 + confidence_level) / 2)  # Z-value for 90% CI
margin_of_error = z_score * sem

ci_lower = sample_mean - margin_of_error
ci_upper = sample_mean + margin_of_error

# Print the results
print("Mean of the sample:", sample_mean)
print("Lower boundary of the 90% CI:", ci_lower)
print("Upper boundary of the 90% CI:", ci_upper)

Mean of the sample: 23.86872
Lower boundary of the 90% CI: 23.804728197025455
Upper boundary of the 90% CI: 23.932711802974545


In [7]:
from scipy.stats import sem, t

# Calculate SEM
sample_sem = sem(sample_close_prices)

# Degrees of freedom
df = len(sample_close_prices) - 1

# Calculate the 90% confidence interval
confidence_level = 0.90
confidence_interval = t.interval(confidence_level, df, loc=sample_mean, scale=sample_sem)

# Results
mean_of_sample = sample_mean
ci_lower = confidence_interval[0]
ci_upper = confidence_interval[1]

mean_of_sample, ci_lower, ci_upper

(np.float64(23.86872),
 np.float64(23.804633053203798),
 np.float64(23.9328069467962))

Mean of the sample: 23.86872

Lower boundary of the 90% CI: 23.804728197025455 

Upper boundary of the 90% CI: 23.932711802974545